hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/federation/FederationStateStoreService.java

```diff
@@ -259,378 +261,190 @@ public Version loadVersion() {
   @Override
   public GetSubClusterPolicyConfigurationResponse getPolicyConfiguration(
       GetSubClusterPolicyConfigurationRequest request) throws YarnException {
-    try {
-      long startTime = clock.getTime();
-      GetSubClusterPolicyConfigurationResponse response =
-          stateStoreClient.getPolicyConfiguration(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("getPolicyConfiguration error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "getPolicyConfiguration", GetSubClusterPolicyConfigurationResponse.class, request);
+    return invoke(clientMethod, GetSubClusterPolicyConfigurationResponse.class);
   }
 
   @Override
   public SetSubClusterPolicyConfigurationResponse setPolicyConfiguration(
       SetSubClusterPolicyConfigurationRequest request) throws YarnException {
-    try {
-      long startTime = clock.getTime();
-      SetSubClusterPolicyConfigurationResponse response =
-          stateStoreClient.setPolicyConfiguration(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("setPolicyConfiguration error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "setPolicyConfiguration", SetSubClusterPolicyConfigurationResponse.class, request);
+    return invoke(clientMethod, SetSubClusterPolicyConfigurationResponse.class);
   }
 
   @Override
   public GetSubClusterPoliciesConfigurationsResponse getPoliciesConfigurations(
       GetSubClusterPoliciesConfigurationsRequest request) throws YarnException {
-    try {
-      long startTime = clock.getTime();
-      GetSubClusterPoliciesConfigurationsResponse response =
-          stateStoreClient.getPoliciesConfigurations(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("getPoliciesConfigurations error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "getPoliciesConfigurations", GetSubClusterPoliciesConfigurationsResponse.class, request);
+    return invoke(clientMethod, GetSubClusterPoliciesConfigurationsResponse.class);
```

Comment:
We have tests for all these metrics, right?

Reply:
I haven't finished the test yet, I will test the interface in junit Test.